future-apps/payroll/contracts/Payroll.sol

```diff
@@ -212,34 +217,79 @@ contract Payroll is AragonApp { //, IForwarder { // makes coverage crash (remove
      * @notice Set employee #`employeeId` annual salary to `denominationSalary` per second.
      * @param employeeId Employee's identifier
      * @param denominationSalary Employee's new salary, per second in denomintation Token
+     * @param changeDate New salary effective date
      */
     function setEmployeeSalary(
         uint128 employeeId,
-        uint256 denominationSalary
+        uint256 denominationSalary,
+        uint64 changeDate
     )
         isInitialized
         employeeExists(employeeId)
         external
         authP(ADD_EMPLOYEE_ROLE, arr(employees[employeeId].accountAddress, denominationSalary, 0))
     {
+        // forbid to change salary in the past
+        require(changeDate >= getTimestamp());
+
+        // Add owed salary to employee's accrued value
+        uint256 owed = _getOwedSalary(employeeId, changeDate);
+
+        employees[employeeId].lastPayroll = changeDate;
+        _addAccruedValue(employeeId, owed);
```

Comment:
This will make all the salary that an employee would be getting during the period between when the function is called and the `changeDate` immediately available, with no option to be taken back if the employee is terminated. A joke such as 'we will raise your salary to 1 USD in 2100' could completely deplete the DAO funds as the employee could get 80 years worth of their previous salary immediately.

We can either remove the possibility to set a custom `changeDate` (and always change at this moment) or add a lot of extra logic for handling different salary periods. I like the former.

Reply:
Good catch! This is so wrong!!

Reply:
It's certainly an edge case, and it wouldn't happen with a proper/regular management, but it's really dangerous and misleading. 
I liked the idea of being able to set the change date in advance and with exactitude, but I guess it can be done right after the date has passed and then adjust the exact amount with `addAccruedValue` function if needed.

Reply:
Another option could be to store last salary change date and not to allow another change if employee hasn't called `payday` after the last one. I guess in regular use cases payday should happen way more often than salary changes. But maybe it's still too much complexity.